src/paperqa/clients/semantic_scholar.py

```diff
@@ -250,27 +253,36 @@ async def s2_title_search(
             HTTPStatus.NOT_FOUND: DOINotFoundError(f"Could not find DOI for {title}.")
         },
     )
+
+    # In case we matched >1, sort by similarity of title
     try:
-        if authors and not s2_authors_match(authors, data=data["data"][0]):
-            raise DOINotFoundError(
-                f"Could not find DOI for {title} - author disagreement."
-            )
-    except KeyError as exc:  # Very rare, but "data" may not be in data
+        title_similarity, result = max(
+            # need to check if nested under a 'data' key or not (depends on filtering)
+            (strings_similarity(entry["title"], title), entry)
+            for entry in data.get("data", data)
+        )
+    except (KeyError, IndexError) as exc:
         raise DOINotFoundError(
             f"Unexpected Semantic Scholar search/match endpoint shape for {title}"
             f" given data {data}."
         ) from exc
-    # need to check if nested under a 'data' key or not (depends on filtering)
+
     if (
-        strings_similarity(
-            data.get("title", "") if "data" not in data else data["data"][0]["title"],
-            title,
-        )
-        < title_similarity_threshold
+        authors
+        and title_similarity < HIGH_TITLE_SIMILARITY_THRESHOLD
+        and not s2_authors_match(authors, data=result)
     ):
+        raise DOINotFoundError(
+            f"Could not find DOI for {title} - author and title disagreement."
+        )
+
+    # If we made it here, either authors were not provided or they matched, so apply
+    # a possibly weakter threshold
+    if title_similarity < title_similarity_threshold:
```

Comment:
Previously, we checked for some disagreement for authors and/or relatively weak title disagreement. Both would fail.

Now, if we have authors - we have a very strict requirement (must be exact match) on title and an author match. Or, we can have no authors and then we have a looser title check. Or, we had an author match and can have a looser title match.

I think my issue is that when we have no authors, I would prefer a tighter title match rather than a looser. Maybe you could break it out into an author/no author if statement. Then you can apply author match + weak or no author match + strict in author branch and strict title in no author branch

Reply:
> if we have authors - we have a very strict requirement (must be exact match) on title and an author match.

If we have authors and we have author match, we fall back to `title_similarity_threshold`. That's how this test case passes (I added a period to break title match): https://github.com/Future-House/paper-qa/blob/5ad5894c4a7fca3383ba0688b6bfc99ce7bfc400/tests/test_clients.py#L520-L524

But your point is taken on no authors - will rewrite this. And try to make it a little more readable.